automation/tests-container-utils.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash -ex
+
+function remove_container {
+    res=$?
+    [ "$res" -ne 0 ] && echo "*** ERROR: $res"
+    container_exec 'rm -rf $CONTAINER_WORKSPACE/*nmstate*.rpm' || true
+    ${CONTAINER_CMD} rm -f $CONTAINER_ID
+}
+
+function container_exec {
+    ${CONTAINER_CMD} exec $USE_TTY -i $CONTAINER_ID \
+        /bin/bash -c "cd $CONTAINER_WORKSPACE && $1"
+}
+
+function open_shell {
+    res=$?
+    [ "$res" -ne 0 ] && echo "*** ERROR: $res"
+    set +o errexit
+    container_exec 'echo "pytest tests/integration --pdb" >> ~/.bash_history'
+    container_exec 'exec /bin/bash'
+    run_exit
+}
+
+# Return 0 if file is changed else 1
+function is_file_changed {
+    git remote add upstream https://github.com/nmstate/nmstate.git
+    git fetch upstream
+    if [ -n "$TRAVIS_BRANCH" ]; then
+        git diff --exit-code --name-only upstream/$TRAVIS_BRANCH -- $1
+    else
+        git diff -exit-code --name-only upstream/master -- $1
+    fi
+
+    if [ $? -eq 0 ];then
+        return 1
+    else
+        return 0
+    fi
+}
+
+function rebuild_container_images {
+    if is_file_changed "$PROJECT_PATH/packaging"; then
+        if [ $CONTAINER_IMAGE == $CENTOS_IMAGE_DEV ];then
+            ${PROJECT_PATH}/packaging/build-container.sh centos8-nmstate-dev
+        elif [ $CONTAINER_IMAGE == $FEDORA_IMAGE_DEV ];then
+            ${PROJECT_PATH}/packaging/build-container.sh fedora-nmstate-dev
+        fi
+    fi
+}
+
+function upgrade_nm_from_copr {
```

Comment:
updating copr might also be interesting for bare-metal or virtual machines.

Reply:
Yes but then this will break compatibility with other systems than Fedora/Centos/RHEL. Could we split this for a future task?

Reply:
Not sure what is breaking here because it requires a command-line switch to runs this code. If you think it is too complicated and you don't need to be able to update VMs, I am fine with delaying this.

Reply:
Copr is not available in other distributions different than Fedora/CentOS/RHEL. @cathay4t wants the baremetal script to be compatible with all the distributions, AFAIK.

Reply:
Since Copr is optional, the script is still compatible as long as the flag is not used.

Reply:
Done, thanks!